src/entry.jl

```
@@ -73,8 +73,17 @@ function publish(branch::AbstractString, prbranch::AbstractString="")
             m !== nothing && occursin(Base.VERSION_REGEX, m.captures[2]) || continue
             pkg, ver = m.captures; ver = VersionNumber(ver)
             sha1 = readchomp(joinpath(metapath,path))
-            old = LibGit2.content(LibGit2.GitBlob(repo, "origin/$branch:$path"))
-            old !== nothing && old != sha1 && throw(Pkg.PkgError("$pkg v$ver SHA1 changed in METADATA – refusing to publish"))
+            try
+                old = LibGit2.content(LibGit2.GitBlob(repo, "origin/$branch:$path"))
+                if old != sha1
+                    throw(Pkg.PkgError("$pkg v$ver SHA1 changed in METADATA – refusing to publish"))
+                end
+            catch e
```

this is exactly what I meant by
> we should avoid needing try catches everywhere if the new api is going to throw in a bunch of expected situations

is there an error code or anything? this will silently swallow all GitErrors, including unexpected ones that should still be fatal.

---

I agree, but this is basically duct tape at this stage. I've tweaked this a bit to be more specific.